Makefile

```diff
@@ -127,3 +129,29 @@ lint_country_dialing_codes: update_pinpoint_supported_countries ## Checks that c
 
 check_asset_strings: ## Checks for strings
 	find ./app/javascript -name "*.js*" | xargs ./scripts/check-assets
+
+build_artifact: $(ARTIFACT_DESTINATION_FILE) ## Builds zipped tar file artifact with IDP source code and Ruby/JS dependencies
+
+$(ARTIFACT_DESTINATION_FILE):
+	@echo "Building artifact into $(ARTIFACT_DESTINATION_FILE)"
+	bundle config set --local cache_all true
+	bundle package
+	tar \
+	  --exclude './config/agencies.yml' \
+	  --exclude './config/iaa_gtcs.yml' \
+	  --exclude './config/iaa_orders.yml' \
+	  --exclude './config/iaa_statuses.yml' \
+	  --exclude './config/integration_statuses.yml' \
+	  --exclude './config/integrations.yml' \
+	  --exclude './config/partner_account_statuses.yml' \
+	  --exclude './config/partner_accounts.yml' \
+	  --exclude './config/service_providers.yml' \
+	  --exclude='./certs/sp' \
+	  --exclude='./identity-idp-config' \
+	  --exclude='./tmp' \
+	  --exclude='./node_modules' \
+	  --exclude='./geo_data/GeoLite2-City.mmdb' \
+	  --exclude='./pwned_passwords/pwned_passwords.txt' \
+	  --exclude='./vendor/ruby' \
+	  --exclude='./config/application.yml' \
+	  -cf - "." | "$(GZIP_COMMAND)" > $@
```

Comment:
sorry for the flip-flop, I was testing this out with the new help documentation in #5792 and we can get the best help message if we 
- combine the targts
- swap out $@ for $(ARTIFACT_DESTINATION_FILE)

```suggestion
build_artifact $(ARTIFACT_DESTINATION_FILE): ## Builds zipped tar file artifact with IDP source code and Ruby/JS dependencies
	@echo "Building artifact into $(ARTIFACT_DESTINATION_FILE)"
	bundle config set --local cache_all true
	bundle package
	tar \
	  --exclude './config/agencies.yml' \
	  --exclude './config/iaa_gtcs.yml' \
	  --exclude './config/iaa_orders.yml' \
	  --exclude './config/iaa_statuses.yml' \
	  --exclude './config/integration_statuses.yml' \
	  --exclude './config/integrations.yml' \
	  --exclude './config/partner_account_statuses.yml' \
	  --exclude './config/partner_accounts.yml' \
	  --exclude './config/service_providers.yml' \
	  --exclude='./certs/sp' \
	  --exclude='./identity-idp-config' \
	  --exclude='./tmp' \
	  --exclude='./node_modules' \
	  --exclude='./geo_data/GeoLite2-City.mmdb' \
	  --exclude='./pwned_passwords/pwned_passwords.txt' \
	  --exclude='./vendor/ruby' \
	  --exclude='./config/application.yml' \
	  -cf - "." | "$(GZIP_COMMAND)" > $(ARTIFACT_DESTINATION_FILE)
```

Reply:
So now it would look like:
```
tmp/idp.tar.gz                      $(ARTIFACT_DESTINATION_FILE) Builds zipped tar file artifact with IDP source code and Ruby/JS dependencies
```